packages/xo-server-audit/src/index.js

```diff
@@ -407,6 +426,74 @@ class AuditXoPlugin {
       }))
   }
 
+  async _handleImportRecords($defer, { fileStream, zipped }) {
+    $defer(await this._storage.acquireLock())
+    if (!(await this._storage.isEmpty())) {
+      throw new Error('Audit log database must be empty')
+    }
+
+    const processRecord = async source => {
+      const invalidRecords = []
+      const missingRecords = []
+      // entries are listed backwards, from newest to oldest. nextLog is the last log we imported
+      let chainLastLog, lastId, nextLog
+      for await (const record of source) {
+        const { id, isValid } = await this._auditCore._importRecord(record)
+        if (lastId === undefined) {
+          lastId = id
+          await this._storage.setLastId(lastId)
+        }
+        if (!isValid) {
+          invalidRecords.push(record.id)
```

Comment:
since it looks you only need the number of invalid record, I think you should use a number instead of an array that can grow infinitely